api/src/main/java/com/datatorrent/api/StorageAgent.java

```diff
@@ -94,4 +94,24 @@
     public void setApplicationAttributes(AttributeMap map);
   }
 
+  interface AsyncStorageAgent
+  {
+    /**
+     * Make checkpoint for given windowID final. i.e after this method returns,
+     * the checkpoint is accessible for recovery.
+     *
+     * @param operatorId
+     * @param windowId
+     * @throws IOException
+     */
+    void finalize(int operatorId, long windowId) throws IOException;
+
+    /**
+     * Check if StorageAgent is configured to take synchronous checkpoints.
+     *
+     * @return true if StorageAgent is configured to take synchronous checkpoints.
+     * @return false otherwise.
+     */
+    boolean isSyncCheckpoint();
```

Comment:
We can't change the base interface as it would make it backward incompatible. and existing StorageAgent needs to be modified (Geode storage agent form Malhar).